CommonTools/RecoAlgos/src/PrimaryVertexAssignment.cc

```diff
@@ -89,13 +107,26 @@ std::pair<int, PrimaryVertexAssignment::Quality> PrimaryVertexAssignment::charge
   // first use "closest in Z" with tight cuts (targetting primary particles)
   const float add_cov = vtxIdMinSignif >= 0 ? vertices[vtxIdMinSignif].covariance(2, 2) : 0.f;
   const float dzE = sqrt(track->dzError() * track->dzError() + add_cov);
-  if (vtxIdMinSignif >= 0 and
+  if (!fOnlyUseFirstDz_ and vtxIdMinSignif >= 0 and
```

Comment:
You can simplify the evaluation if you consider separately the two cases for `fOnlyUseFirstDz_`:
```
if (!fOnlyUseFirstDz_) {
  ...
} else {
   ...
}
```